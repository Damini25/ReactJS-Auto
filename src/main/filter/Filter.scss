.filter-main{
    width: 30%;
    height: 100%;
    padding: 18px;
    .subdiv{
        height: 40%;
        border: 2px solid #ededed;
        border-radius: 2px;
        padding: 0 24px;
        .control{
            padding: 12px 0px;
        }
        label{
            font-size: 12px;
        }
        select{
            width: 100%;
            height: 40px;
            margin-top: 7px;
            border-color: #ededed;
        }
        .btndiv{
            text-align: right;
            margin: 10% 0;
        }
        .filterbtn{
            width: 30%;
            height: 32px;
            background: #EA7F28;
            color: white;
            border: none;
        }
    }
}