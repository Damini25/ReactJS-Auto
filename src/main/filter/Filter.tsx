import React from 'react';
import './Filter.scss';

export interface Props{
  renderColorOptions:()=>void,
  renderManufacturerOptions:()=>void,
  onInputChange:(evt:React.ChangeEvent<HTMLSelectElement>) => void,
  selectedColor:string,
  selectedManufacturer:string
  onFilter:() => void;
}
function Filter({renderManufacturerOptions,renderColorOptions,onInputChange,selectedColor="none",selectedManufacturer="none",onFilter}:Props) {
  
  const handleChange = (e:React.ChangeEvent<HTMLSelectElement>) => {
    onInputChange(e);  
  }

  return (
    <div className="filter-main">
      <div className="subdiv">
        <div className="control">
          <label>Color</label>
          <div>
            <select
              value={selectedColor}
              onChange={(e)=>handleChange(e)}
              name="color"
              data-testid="color-test-id"
            >
              <option value="none">All car colors</option>
              {renderColorOptions()}
            </select>
          </div>

        </div>
        <div className="control">
          <label>
            Manufacturer
          </label>
          <div>
            <select
              value={selectedManufacturer}
              onChange={(e)=>handleChange(e)}
              name="manufacturer"
              data-testid="manufacturer-test-id"
            >
              <option value="none">All manufacturers</option>
             {renderManufacturerOptions()}
            </select>
          </div>

        </div>
        <div className="btndiv"> <button className="filterbtn" onClick={onFilter}>Filter</button></div>
      </div>
    </div>
  );
}

export default Filter;