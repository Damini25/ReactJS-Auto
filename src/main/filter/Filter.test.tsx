import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Filter from './Filter';

const props = {
    renderColorOptions:jest.fn,
    renderManufacturerOptions:jest.fn,
    selectedColor: "none",
    selectedManufacturer: "none",
    onFilter:jest.fn(),
    onInputChange:jest.fn,
}
afterEach(cleanup)
test('render <Filter>', () => {
    const wrapper = render(<Filter {...props} />)
    expect(wrapper).toMatchSnapshot()
})
it('it should change the value on changing color options', () => {
    const wrapper = render(<Filter {...props} />); 
    const colorInput = wrapper.getByTestId('color-test-id')as HTMLSelectElement
    const event = {
        target: { value: "none" }
      } as React.ChangeEvent<HTMLSelectElement>;
    fireEvent.change(colorInput,event)
    expect(colorInput.value).toBe('none')
});
it('it should change the value on changing manufacturer options', () => {
    const wrapper = render(<Filter {...props} />); 
    const manfInput = wrapper.getByTestId('manufacturer-test-id') as HTMLSelectElement
    const event = {
        target: { value: "none" }
      } as React.ChangeEvent<HTMLSelectElement>;
    fireEvent.change(manfInput,event)
    expect(manfInput.value).toBe('none')
});