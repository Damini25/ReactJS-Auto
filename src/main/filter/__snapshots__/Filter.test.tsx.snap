// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Filter> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filter-main"
      >
        <div
          class="subdiv"
        >
          <div
            class="control"
          >
            <label>
              Color
            </label>
            <div>
              <select
                data-testid="color-test-id"
                name="color"
              >
                <option
                  value="none"
                >
                  All car colors
                </option>
              </select>
            </div>
          </div>
          <div
            class="control"
          >
            <label>
              Manufacturer
            </label>
            <div>
              <select
                data-testid="manufacturer-test-id"
                name="manufacturer"
              >
                <option
                  value="none"
                >
                  All manufacturers
                </option>
              </select>
            </div>
          </div>
          <div
            class="btndiv"
          >
             
            <button
              class="filterbtn"
            >
              Filter
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filter-main"
    >
      <div
        class="subdiv"
      >
        <div
          class="control"
        >
          <label>
            Color
          </label>
          <div>
            <select
              data-testid="color-test-id"
              name="color"
            >
              <option
                value="none"
              >
                All car colors
              </option>
            </select>
          </div>
        </div>
        <div
          class="control"
        >
          <label>
            Manufacturer
          </label>
          <div>
            <select
              data-testid="manufacturer-test-id"
              name="manufacturer"
            >
              <option
                value="none"
              >
                All manufacturers
              </option>
            </select>
          </div>
        </div>
        <div
          class="btndiv"
        >
           
          <button
            class="filterbtn"
          >
            Filter
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
